valhalla/baldr/traffictile.h

```diff
@@ -0,0 +1,177 @@
+#ifndef VALHALLA_BALDR_TRAFFICTILE_H_
+#define VALHALLA_BALDR_TRAFFICTILE_H_
+
+// This header supports exporting as a set of C++ minimal structs
+// with namespaces, but with C-99 only types and headers, to make
+// binding generation in other languages simpler.
+// If you do `-DC_ONLY_INTERFACE`, you'll get a header with
+// C99 stdint.h, and POD structs with no constructors
+#ifndef C_ONLY_INTERFACE
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <vector>
+#else
+#include <stdint.h>
+#endif
+
+#ifndef C_ONLY_INTERFACE
+namespace valhalla {
+namespace baldr {
+namespace traffic {
+
+using std::uint16_t;
+using std::uint32_t;
+using std::uint64_t;
+#endif
+
+struct Speed {
+  uint16_t speed_kmh : 7;        // km/h - so max range is 0-127km/h
+  uint16_t congestion_level : 3; // some value from 0 to 7 to report back
+  uint16_t is_scale : 1;         // treat speed as a floating point multiplier to edge speed
+  uint16_t age : 4;              //
+  uint16_t has_incident : 1;     // TODO: reserved for whether edge has an incident or not
+
+#ifndef C_ONLY_INTERFACE
+  Speed() : speed_kmh{0}, congestion_level{0}, is_scale{0}, age{0}, has_incident{0} {
+  }
+  Speed(const Speed& other) = default;
+  Speed(const volatile Speed& other)
+      : speed_kmh{other.speed_kmh}, congestion_level{other.congestion_level},
+        is_scale{other.is_scale}, age{other.age}, has_incident{other.has_incident} {
+  }
+#endif
+};
+
+// single incident record
+struct Incident {
+  uint64_t edge_index : 21;
+  uint64_t incident_type : 8;
+  uint64_t start_location : 10;
+  uint64_t length : 10;
+  uint64_t spare : 15;
+#ifndef C_ONLY_INTERFACE
+  Incident() : edge_index{0}, incident_type{0}, start_location{0}, length{0}, spare{0} {
+  }
+  Incident(const Incident& other) = default;
+  Incident(const volatile Incident& other)
+      : edge_index{other.edge_index}, incident_type{other.incident_type},
+        start_location{other.start_location}, length{other.length}, spare{0} {
+  }
+  Incident& operator=(const Incident& other) = default;
+#endif
+};
+
+// per-speed-tile header
+struct TileHeader {
+  uint64_t tile_id;
+  uint32_t directed_edge_count;
+  uint32_t incident_buffer_size;
```

Comment:
instead of backbuffer frontbuffer swapping and having the speed tiles reference the incidents, why not decouple them completely? we could have incidents be in their own tiles with their own refresh rate and delivery to process mechanism. since they are variable size but still very small maybe we can just mount a ram disk and reload them from that disk on some interval. we dont need them to make decisions about expansion in a tight loop (ie the use for incidents will be in `triplegbuilder.cc` where we would only need to refresh those incident tiles along our path) the requirements are quite a bit less. obviously the binary search with the sorted incidents makes perfect sense but we should leave the mechanism by which we access the incidents out until we have a fully fleshed out plan for that

Reply:
Should I just rip out incident support for now then, and we'll add it later when we're ready?